### PR TITLE
Added new topic about Magento's software lifecycle policy 

### PR DESCRIPTION
**Pull request by jeff-matthews**
## Purpose of this pull request

This pull request (PR) adds a new subsection to the Upcoming releases section called “Lifecycle Policy”.

## Affected DevDocs pages

- https://devdocs.magento.com/release/lifecycle-policy.html

whatsnew
Added a new Magento [Lifecycle policy](https://devdocs.magento.com/release/lifecycle-policy.html) topic.

### CODE REVIEW

File: src/release/lifecycle-policy.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+For {{site.data.var.ece}} 2.3 and subsequent releases:
+
+-  Magento will provide quality fixes for a minor release for a minimum of 12 months from the general availability announcement date of the next minor software release.
+
+-  Magento will provide security fixes for a minor release for a minimum of 18 months from the general availability announcement date of the next minor software release.
+
+-  Magento will provide security fixes through patch releases for currently supported minor release versions only. Security fixes will not be backported to previous minor releases, nor to previous patch releases within supported minor releases.
+
+   For example, while 2.3 and 2.4 are currently supported minor releases, security fixes will be released as 2.3.X and 2.4.Y, where X and Y represent the next incremental patch release cumulative of all prior patches. Patch releases will not be released for prior minor releases that have reached end of support (for example, 2.2.Z), or as patches to prior patch releases (for example, 2.3.1.X).
```
Comment (BarnyShergold) [inline src/release/lifecycle-policy.md:14]: Suggested as this is how these patch releases appear on documentation & site. For me the 'p' adds clarity that this is a patch release.
```suggestion
   For example, while 2.3 and 2.4 are currently supported minor releases, security fixes will be released as 2.3.X and 2.4.Y, where X and Y represent the next incremental patch release cumulative of all prior patches. Patch releases will not be released for prior minor releases that have reached end of support (for example, 2.2.Z), or as patches to prior patch releases (for example, 2.3.1-pX).
```

File: src/release/lifecycle-policy.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+   For example, while 2.3 and 2.4 are currently supported minor releases, individual patches will be released as 2.3.X and release patches will be released as 2.4.Y, where X and Y represent the next incremental patch release cumulative of all prior patches. Neither patch releases nor individual patches will be released for prior minor releases that have reached end of support (e.g. 2.2.2).
+
+| Release                   | Release Date  | End of Quality Fixes | End of Security Fixes/<br>End of Software Support |
+|---------------------------|---------------|----------------------|---------------------------------------------------|
+| {{site.data.var.ece}} 2.3 | November 2018 | July 2021            | April 2023<sup>*</sup>                            |
+| {{site.data.var.ece}} 2.4 | July 2020     |                      |                                                   |
+
+*<sup>*</sup> The End of Software Support date for {{site.data.vr.ece}} 2.3 has been extended to April 2022 due to impacts from COVID-19.*
```
Comment (BarnyShergold) [inline src/release/lifecycle-policy.md:25]: Comment says 2022 but table says 2023

**Review by BarnyShergold — CHANGES_REQUESTED** [review]

File: src/release/lifecycle-policy.md (hunk 1 above)
Comment (coreydulimba-ad) [reply src/release/lifecycle-policy.md:14]: Thanks for the suggestion the issue with adding a -p is that then it signifies our security only line which could add confusion.

File: src/release/lifecycle-policy.md (hunk 2 above)
Comment (jeff-matthews) [reply src/release/lifecycle-policy.md:25]: Nice catch, Thanks @BarnyShergold. Changed the table date to 2022 to match the lifecycle policy PDF: https://magento.com/sites/default/files/magento-software-lifecycle-policy.pdf

File: src/release/lifecycle-policy.md (hunk 1 above)
Comment (BarnyShergold) [reply src/release/lifecycle-policy.md:14]: But I would say that having something like 2.3.1.1 (as 2.3.1.X indicates) is also confusing as Magento only use 2.X.Y numbers. A 4th number segment never appears anywhere else. What do you think? If you think it's fine then I won't contest  😄

Comment (BarnyShergold) [reply src/release/lifecycle-policy.md:14]: @jeff-matthews - What do you think?

Comment (jeff-matthews) [reply src/release/lifecycle-policy.md:14]: I'm leaning towards what @coreydulimba-ad said @BarnyShergold.

Comment (BarnyShergold) [reply src/release/lifecycle-policy.md:14]: @jeff-matthews Okay- as the consensus is to leave as is, I'll agree

**Review by BarnyShergold — APPROVED** [review]

File: src/release/lifecycle-policy.md (hunk 1 above)
Comment (jeff-matthews) [reply src/release/lifecycle-policy.md:14]: Thanks @BarnyShergold!

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**